libavfilter/dnn/dnn_backend_tf.c

```diff
@@ -351,11 +351,13 @@ static DNNReturnType add_conv_layer(TFModel *tf_model, TF_Operation *transpose_o
     memcpy(TF_TensorData(tensor), params->kernel, size * sizeof(float));
     TF_SetAttrTensor(op_desc, "value", tensor, tf_model->status);
     if (TF_GetCode(tf_model->status) != TF_OK){
+        TF_DeleteTensor(tensor);
         av_log(ctx, AV_LOG_ERROR, "Failed to set value for kernel of conv layer %d\n", layer);
         return DNN_ERROR;
     }
     op = TF_FinishOperation(op_desc, tf_model->status);
     if (TF_GetCode(tf_model->status) != TF_OK){
+        TF_DeleteTensor(tensor);
```

Comment:
also need at line 375

Reply:
and line 390

Reply:
Line 375 for adding node "Transpose". Since no mem allocated for it (but mem allocated in line 350 for node conv_kernel), i was thinking if it failed, maybe no mem free needed.  
But after think twice, if error occurs in adding some node, all the allocated tensor before should be free, right?
exp: if there is error in line 408, so both allocated mem in line 399/350 should be free? Am I right or overthinking?

Reply:
yes, looks that we need a place to free all the mem in the mode.

maybe you can add error handling at the end of this function, and use 'goto error' upon error in the function

Reply:
Yes, you're right. I added 'err' at the end of this func and use 'goto err' when error occurs.
BTW, all TF_DeleteXXX(...) accept NULL pointer as input.